osu.Game/Overlays/Settings/Sections/Maintenance/StableDirectorySelectScreen.cs

```diff
@@ -15,7 +15,16 @@ public partial class StableDirectorySelectScreen : DirectorySelectScreen
 
         protected override OverlayActivation InitialOverlayActivationMode => OverlayActivation.Disabled;
 
-        protected override bool IsValidDirectory(DirectoryInfo? info) => info?.GetFiles("osu!.*.cfg").Any() ?? false;
+        protected override bool IsValidDirectory(DirectoryInfo? info) =>
+            // A full stable installation will have a configuration file present.
+            // This is the best case scenario, as it may contain a custom beatmap directory we need to traverse to.
+            info?.GetFiles("osu!.*.cfg").Any() == true ||
+            // The user may only have their songs or skins folders left.
+            // We still want to allow them to import based on this.
+            info?.GetDirectories("Songs").Any() == true ||
+            info?.GetDirectories("Skins").Any() == true ||
+            // The user may have traverse *inside* their songs or skins folders.
+            shouldUseParentDirectory(info);
```

Comment:
And we are okay with this meaning that the user can potentially import any old garbage folder, so long as it includes a `Songs` or `Skins` directory, yeah? And also that any folder named `Skins` or `Songs` - even if it is not actually a skins or songs folder - will always be importable and will implicitly assume its parent is a root of a stable install, even though it may not necessarily be so?

Yes for the happy paths this will do sensible things, but I'm also attempting to examine the unhappy paths wherein it might very much not.

Reply:
I think it should be fine. The actual importers should filter out garbage data, is the hope.

Reply:
Okay then. I will not attempt to break the importers as I assume the users will be better at doing so anyways.